README
Y 0.2
=====

Visit http://www.y-windows.org/ for more information.



BUILDING
--------

If you are using a development snapshot, or got this tree from an Arch
archive, then you need to run

  ./autogen.sh

before the tree will be usable.

To configure, compile and install, use:

  ./configure --prefix=/path/to/install/dir
  make
  make install

You MUST install before Y will run as it needs to locate its modules in the
installed directory.


# arch-tag: 636c673b-8186-440d-be33-e76a54f05ee9


ChangeLog
# do not edit -- automatically generated by arch changelog
# arch-tag: automatic-ChangeLog--markbt@efaref.net--2004/Y--devel--0.2
#

2004-05-21 17:19:09 GMT	Mark B. Thomas <markbt@efaref.net>	patch-51

    Summary:
      merge with asuffield
    Revision:
      Y--devel--0.2--patch-51

    merge with asuffield
    
    Patches applied:
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-93
       begin implementing ykbcomp
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-94
       implement some more of ykbcomp
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-95
       finish ykbcomp
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-96
       add 'clear' action
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-97
       implement rbtree
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-98
       add rbtree_check, fix rbtree
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-99
       fix a few memory leaks
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-100
       implement direct-mapped method arguments, and rip out all useless translation functions
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-101
       fix a few more leaks
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-102
       merge with markbt
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-103
       add branch changelog
    

    new files:
     ChangeLog.d/asuffield@debian.org--gluck-2004
     Y/util/.arch-ids/rbtree.c.id Y/util/.arch-ids/rbtree.h.id
     Y/util/rbtree.c Y/util/rbtree.h Y/util/rbtree_check.c
     config/keymaps/.arch-ids/ykbcomp.in.id
     config/keymaps/YKB/.arch-ids/=id
     config/keymaps/YKB/.arch-ids/YKB.pm.id
     config/keymaps/YKB/.arch-ids/YKBGrammar.id
     config/keymaps/YKB/.arch-ids/YKM.pm.id
     config/keymaps/YKB/YKB.pm config/keymaps/YKB/YKBGrammar
     config/keymaps/YKB/YKM.pm config/keymaps/ykbcomp.in

    modified files:
     ChangeLog.d/markbt@efaref.net--2004 Y/Makefile.am
     Y/input/keymap.c Y/main/control.c Y/main/y.c
     Y/message/client.c Y/message/client.h Y/message/message.c
     Y/message/message.h Y/util/llist.h Y/util/log.c Y/util/log.h
     Y/widget/button.c Y/widget/canvas.c Y/widget/checkbox.c
     Y/widget/console.c Y/widget/desktop.c Y/widget/gridlayout.c
     Y/widget/gridlayout.h Y/widget/label.c Y/widget/menu.c
     Y/widget/window.c Y/widget/window.h build-misc/yclpp
     config/keymaps/Makefile.am config/keymaps/gb.ykb

    new directories:
     config/keymaps/YKB config/keymaps/YKB/.arch-ids

    new patches:
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-93
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-94
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-95
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-96
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-97
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-98
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-99
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-100
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-101
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-102
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-103


2004-05-21 16:40:21 GMT	Mark B. Thomas <markbt@efaref.net>	patch-50

    Summary:
      merge Y--pixelbuffer-0.2
    Revision:
      Y--devel--0.2--patch-50

    merge Y--pixelbuffer-0.2
    
    Patches applied:
    
     * markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-6
       merge with Y--devel--0.2
    
     * markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-7
       add painterDrawBuffer
    
     * markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-8
       merge with Y--devel--0.2
    

    modified files:
     ChangeLog.d/markbt@efaref.net--2004 Y/buffer/buffer.c
     Y/buffer/buffer.h Y/buffer/bufferclass.h Y/buffer/painter.c
     Y/buffer/painter.h Y/buffer/painterclass.h
     Y/buffer/rgbabuffer.c Y/input/pointer.c Y/input/pointer.h

    new patches:
     markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-6
     markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-7
     markbt@efaref.net--2004/Y--pixelbuffer--0.2--patch-8


2004-05-16 18:06:00 GMT	Mark B. Thomas <markbt@efaref.net>	patch-49

    Summary:
      merge with asuffield
    Revision:
      Y--devel--0.2--patch-49

    merge with asuffield
    
    Patches applied:
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-90
       merge with markbt
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-91
       fix a few ykb bugs
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-92
       add stroke description and notification to ykb
    

    modified files:
     ChangeLog.d/markbt@efaref.net--2004 Y/Makefile.am
     Y/input/ykb.c Y/input/ykb.h Y/input/ykbmap.c
     Y/input/ykbmap_p.h Y/widget/widget.c Y/widget/widget.h
     Y/widget/widget_p.h config/keymaps/gb.ykb

    new patches:
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-90
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-91
     asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-92


2004-05-15 21:16:34 GMT	Mark B. Thomas <markbt@efaref.net>	patch-48

    Summary:
      merge with asuffield
    Revision:
      Y--devel--0.2--patch-48

    merge with asuffield
    
    Patches applied:
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-88
       fix the handling of edge combining in ykbmap.c
    
     * asuffield@debian.org--gluck-2004/Y--asuffield--0.2--patch-89
       merge from markbt
    

    modified files:
     ChangeLog.d/markbt@efaref.net--2004 Y/input/ykbmap.c
     config/keymaps/gb.ykb

    new patches:

Y/const.h
#ifndef Y_CONST_H
#define Y_CONST_H

/* Y constants */

enum YMessageOperation
{
  YMO_ERROR,
  YMO_QUIT,
  YMO_EVENT,
  YMO_FIND_CLASS,
  YMO_INVOKE_CLASS_METHOD,
  YMO_INVOKE_INSTANCE_METHOD,
};

/* The Keyboard Codes are cunningly taken from SDL, since that seems
 * to be a fairly sane way of doing it.
 */

/* What we really want is a mapping of every raw key on the keyboard.
   To support international keyboards, we use the range 0xA1 - 0xFF
   as international virtual keycodes.  We'll follow in the footsteps of X11...
   The names of the keys
 */
 
enum YKeyCode
{
	/* The keyboard syms have been cleverly chosen to map to ASCII */
	YK_UNKNOWN		= 0,
	YK_FIRST		= 0,
        YK_CTRL_A               = 1,
        YK_CTRL_B               = 2,
        YK_CTRL_C               = 3,
        YK_CTRL_D               = 4,
        YK_CTRL_E               = 5,
        YK_CTRL_F               = 6,
        YK_CTRL_G               = 7,
        YK_CTRL_H               = 8,
	YK_BACKSPACE		= 8,
        YK_CTRL_I               = 9,
	YK_TAB			= 9,
        YK_CTRL_J               = 10,
        YK_CTRL_K               = 11,
        YK_CTRL_L               = 12,
	YK_CLEAR		= 12,
        YK_CTRL_M               = 13,
	YK_RETURN		= 13,
        YK_CTRL_N               = 14,
        YK_CTRL_O               = 15,
        YK_CTRL_P               = 16,
        YK_CTRL_Q               = 17,
        YK_CTRL_R               = 18,
        YK_CTRL_S               = 19,
	YK_PAUSE		= 19,
        YK_CTRL_T               = 20,
        YK_CTRL_U               = 21,
        YK_CTRL_V               = 22,
        YK_CTRL_W               = 23,
        YK_CTRL_X               = 24,
        YK_CTRL_Y               = 25,
        YK_CTRL_Z               = 26,
        YK_CTRL_LEFTBRACKET     = 27,
	YK_ESCAPE		= 27,
        YK_CTRL_BACKSLASH       = 28,
        YK_CTRL_RIGHTBRACKET    = 29,
        YK_CTRL_CARET           = 30,
        YK_CTRL_UNDERSCORE      = 31,
	YK_SPACE		= 32,
	YK_EXCLAIM		= 33,
	YK_QUOTEDBL		= 34,
	YK_HASH			= 35,
	YK_DOLLAR		= 36,
        YK_PERCENT              = 37,
	YK_AMPERSAND		= 38,
	YK_QUOTE		= 39,
	YK_LEFTPAREN		= 40,
	YK_RIGHTPAREN		= 41,
	YK_ASTERISK		= 42,
	YK_PLUS			= 43,
	YK_COMMA		= 44,
	YK_MINUS		= 45,
	YK_PERIOD		= 46,
	YK_SLASH		= 47,
	YK_0			= 48,
	YK_1			= 49,
	YK_2			= 50,
	YK_3			= 51,
	YK_4			= 52,
	YK_5			= 53,
	YK_6			= 54,
	YK_7			= 55,
	YK_8			= 56,
	YK_9			= 57,
	YK_COLON		= 58,
	YK_SEMICOLON		= 59,
	YK_LESS			= 60,
	YK_EQUALS		= 61,
	YK_GREATER		= 62,
	YK_QUESTION		= 63,
	YK_AT			= 64,
        YK_A                    = 65,
        YK_B                    = 66,
        YK_C                    = 67,
        YK_D                    = 68,
        YK_E                    = 69,
        YK_F                    = 70,
        YK_G                    = 71,
        YK_H                    = 72,
        YK_I                    = 73,
        YK_J                    = 74,
        YK_K                    = 75,
        YK_L                    = 76,
        YK_M                    = 77,
        YK_N                    = 78,
        YK_O                    = 79,
        YK_P                    = 80,
        YK_Q                    = 81,
        YK_R                    = 82,
        YK_S                    = 83,
        YK_T                    = 84,
        YK_U                    = 85,
        YK_V                    = 86,
        YK_W                    = 87,
        YK_X                    = 88,
        YK_Y                    = 89,
        YK_Z                    = 90,
	YK_LEFTBRACKET		= 91,
	YK_BACKSLASH		= 92,
	YK_RIGHTBRACKET		= 93,
	YK_CARET		= 94,
	YK_UNDERSCORE		= 95,
	YK_BACKQUOTE		= 96,
	YK_a			= 97,
	YK_b			= 98,
	YK_c			= 99,
	YK_d			= 100,
	YK_e			= 101,
	YK_f			= 102,
	YK_g			= 103,
	YK_h			= 104,
	YK_i			= 105,
	YK_j			= 106,
	YK_k			= 107,
	YK_l			= 108,
	YK_m			= 109,
	YK_n			= 110,
	YK_o			= 111,
	YK_p			= 112,
	YK_q			= 113,
	YK_r			= 114,
	YK_s			= 115,
	YK_t			= 116,
	YK_u			= 117,
	YK_v			= 118,
	YK_w			= 119,
	YK_x			= 120,
	YK_y			= 121,
	YK_z			= 122,
        YK_LEFTBRACE            = 123,
        YK_PIPE                 = 124,
        YK_RIGHTBRACE           = 125,
        YK_TILDE                = 126,
	YK_DELETE		= 127,
	/* End of ASCII mapped keysyms */

	/* International keyboard syms */
	YK_WORLD_0		= 160,		/* 0xA0 */
	YK_WORLD_1		= 161,
	YK_WORLD_2		= 162,
	YK_WORLD_3		= 163,
	YK_WORLD_4		= 164,
	YK_WORLD_5		= 165,
	YK_WORLD_6		= 166,
	YK_WORLD_7		= 167,
	YK_WORLD_8		= 168,
	YK_WORLD_9		= 169,
	YK_WORLD_10		= 170,
	YK_WORLD_11		= 171,
	YK_WORLD_12		= 172,
	YK_WORLD_13		= 173,
	YK_WORLD_14		= 174,
	YK_WORLD_15		= 175,
	YK_WORLD_16		= 176,
	YK_WORLD_17		= 177,
	YK_WORLD_18		= 178,
	YK_WORLD_19		= 179,
	YK_WORLD_20		= 180,
	YK_WORLD_21		= 181,
	YK_WORLD_22		= 182,
	YK_WORLD_23		= 183,
	YK_WORLD_24		= 184,
	YK_WORLD_25		= 185,
	YK_WORLD_26		= 186,
	YK_WORLD_27		= 187,
	YK_WORLD_28		= 188,
	YK_WORLD_29		= 189,
	YK_WORLD_30		= 190,
	YK_WORLD_31		= 191,
	YK_WORLD_32		= 192,
	YK_WORLD_33		= 193,
	YK_WORLD_34		= 194,
	YK_WORLD_35		= 195,
	YK_WORLD_36		= 196,
	YK_WORLD_37		= 197,
	YK_WORLD_38		= 198,
	YK_WORLD_39		= 199,
	YK_WORLD_40		= 200,
	YK_WORLD_41		= 201,
	YK_WORLD_42		= 202,
	YK_WORLD_43		= 203,
	YK_WORLD_44		= 204,
	YK_WORLD_45		= 205,
	YK_WORLD_46		= 206,
	YK_WORLD_47		= 207,
	YK_WORLD_48		= 208,
	YK_WORLD_49		= 209,
	YK_WORLD_50		= 210,
	YK_WORLD_51		= 211,
	YK_WORLD_52		= 212,
	YK_WORLD_53		= 213,
	YK_WORLD_54		= 214,
	YK_WORLD_55		= 215,
	YK_WORLD_56		= 216,
	YK_WORLD_57		= 217,
	YK_WORLD_58		= 218,
	YK_WORLD_59		= 219,
	YK_WORLD_60		= 220,
	YK_WORLD_61		= 221,
	YK_WORLD_62		= 222,
	YK_WORLD_63		= 223,
	YK_WORLD_64		= 224,
	YK_WORLD_65		= 225,
	YK_WORLD_66		= 226,
	YK_WORLD_67		= 227,
	YK_WORLD_68		= 228,
	YK_WORLD_69		= 229,
	YK_WORLD_70		= 230,
	YK_WORLD_71		= 231,
	YK_WORLD_72		= 232,
	YK_WORLD_73		= 233,
	YK_WORLD_74		= 234,
	YK_WORLD_75		= 235,
	YK_WORLD_76		= 236,
	YK_WORLD_77		= 237,
	YK_WORLD_78		= 238,
	YK_WORLD_79		= 239,
	YK_WORLD_80		= 240,
	YK_WORLD_81		= 241,
	YK_WORLD_82		= 242,
	YK_WORLD_83		= 243,
	YK_WORLD_84		= 244,
	YK_WORLD_85		= 245,
	YK_WORLD_86		= 246,
	YK_WORLD_87		= 247,
	YK_WORLD_88		= 248,
	YK_WORLD_89		= 249,
	YK_WORLD_90		= 250,
	YK_WORLD_91		= 251,
	YK_WORLD_92		= 252,
	YK_WORLD_93		= 253,
	YK_WORLD_94		= 254,
	YK_WORLD_95		= 255,		/* 0xFF */

	/* Numeric keypad */
	YK_KP0			= 256,
	YK_KP1			= 257,
	YK_KP2			= 258,
	YK_KP3			= 259,
	YK_KP4			= 260,
	YK_KP5			= 261,
	YK_KP6			= 262,
	YK_KP7			= 263,
	YK_KP8			= 264,
	YK_KP9			= 265,
	YK_KP_PERIOD		= 266,
	YK_KP_DIVIDE		= 267,
	YK_KP_MULTIPLY		= 268,
	YK_KP_MINUS		= 269,
	YK_KP_PLUS		= 270,
	YK_KP_ENTER		= 271,
	YK_KP_EQUALS		= 272,

	/* Arrows + Home/End pad */
	YK_UP			= 273,
	YK_DOWN			= 274,
	YK_RIGHT		= 275,
	YK_LEFT			= 276,
	YK_INSERT		= 277,
	YK_HOME			= 278,
	YK_END			= 279,
	YK_PAGEUP		= 280,
	YK_PAGEDOWN		= 281,

	/* Function keys */
	YK_F1			= 282,
	YK_F2			= 283,
	YK_F3			= 284,
	YK_F4			= 285,
	YK_F5			= 286,
	YK_F6			= 287,
	YK_F7			= 288,
	YK_F8			= 289,
	YK_F9			= 290,
	YK_F10			= 291,
	YK_F11			= 292,
	YK_F12			= 293,
	YK_F13			= 294,
	YK_F14			= 295,
	YK_F15			= 296,

	/* Key state modifier keys */
	YK_NUMLOCK		= 300,
	YK_CAPSLOCK		= 301,
	YK_SCROLLLOCK		= 302,
	YK_RSHIFT		= 303,
	YK_LSHIFT		= 304,
	YK_RCTRL		= 305,
	YK_LCTRL		= 306,
	YK_RALT			= 307,
	YK_LALT			= 308,
	YK_RMETA		= 309,
	YK_LMETA		= 310,
	YK_LSUPER		= 311,		/* Left "Windows" key */
	YK_RSUPER		= 312,		/* Right "Windows" key */
	YK_MODE			= 313,		/* "Alt Gr" key */
	YK_COMPOSE		= 314,		/* Multi-key compose key */

	/* Miscellaneous function keys */
	YK_HELP			= 315,
	YK_PRINT		= 316,
	YK_SYSREQ		= 317,
	YK_BREAK		= 318,
	YK_MENU			= 319,
	YK_POWER		= 320,		/* Power Macintosh power key */
	YK_EURO			= 321,		/* Some european keyboards */

	/* Add any other keys here */

	YK_LAST
};

/* Enumeration of valid key mods (possibly OR'd together) */
enum YModifierCode
{
  YMOD_NONE    = 0x0000,
  YMOD_SHIFT   = 0x0001,
  YMOD_CTRL    = 0x0002,
  YMOD_ALT     = 0x0004,
  YMOD_META    = 0x0010,
  YMOD_MODE    = 0x0020,
  YMOD_NUMLOCK = 0x0040
};


#endif /* header guard */

/* arch-tag: 018dc291-69c6-4595-aa83-ba1a12ab326c
 */
